addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -98,11 +98,15 @@ function factory(dependencies) {
          */
         async remove() {
             if (!this.isTemporary) {
+                const emptyMessages = this.messages.filter(message =>
+                    !message.body || message.body.replace(/<\/?[^>]+>/g, '').trim() === ''
+                );
```

Comment:
If possible this should come strictly from the server (see comment about using bus).

Reply:
This would fail to remove the message from other tabs.